lms/services/lti_grading/_v13.py

```diff
@@ -44,6 +58,18 @@ def read_result(self, grading_id) -> GradingResult:
             raise
 
         results = response.json()
+        if self._product_family == Family.BLACKBOARD:
+            # Due to the bug mentioned above we didn't use the API filtering by user for blackboard.
+            # We'll filter ourselves here instead.
+            results = [result for result in results if result["userId"] == grading_id]
+
+            # On top of this, blackboard adds some HTML to the comments it returns
+            _ = [
+                result.update(comment=strip_html_tags(result["comment"]))
+                for result in results
+                if result.get("comment")
+            ]
```

Comment:
Maybe we can do a regular loop here, as it's easier to reason about (I had to read twice to get what was the underscore var for)

```suggestion
            for result in results:
                if result.get("comment"):
                    result.update(comment=strip_html_tags(result["comment"]))
```

Reply:
I still agree this was better but it needs to:

- Use a different name for `result`, eg `result_` to avoid masking the outer scope one
- A pylint:disable for growing complexity (because of the new if)


I'm going to revert to the list comprehension.